libs/os/dir.h

```diff
@@ -24,24 +24,40 @@
 /// \file
 /// \brief OS directory-listing object.
 
-#include <glib.h>
-
-typedef GDir Directory;
-
-inline bool directory_good( Directory* directory ){
-	return directory != 0;
+class Directory {
+	private:
+		int entry_idx = 0;
+		void* dir;
+
+	public:
+		Directory(const char* name);
+		bool good();
+		void close();
+		const char* read_and_increment();
+};
+
+inline bool directory_good(Directory* directory) {
+	if(directory) {
```

Comment:
Design with two pointer null checks is semantically excess (Directory* and QDir*). Directory never gets null practically and may be of non pointer type.

Reply:
changed